lib/commands/context.js

```diff
@@ -192,8 +194,11 @@ async function setupExistingChromedriver (chromedriver) {
 }
 
 async function setupNewChromedriver (opts, curDeviceId, adbPort) {
+  let getPort = BlueBird.promisify(PortFinder.getPort, {context: PortFinder});
+  let chromePort = await getPort();
+  logger.error(`Using ${chromePort} for new ChromeDriver session`);
   let chromeArgs = {
-    port: opts.chromeDriverPort,
+    port: chromePort,
```

Comment:
Something I noticed: In the logs, I see Appium tries to kill any old existing Chromedriver processes:
```
info Chromedriver Killing any old chromedrivers, running: pkill -15 -f "/Users/parks169/code/appium-android-driver/node_modules/appium-chromedriver/chromedriver/mac/chromedriver.*--port=9515"
```

The port is included in the `killAll` command. So if a Chromedriver process leaks with a random port, it will not be caught by this code.

One option is to remove the port in the command. This can introduce the problem of inadvertently killing other Chromedriver processes that are running but not spawned from Appium.

Another option is to base the `killAll` on the parent process/group. But I think if a previous Appium process leaked a Chromedriver process, our new Appium process won't be able to tell. It's just another PID number without any more information.

Maybe `killAll` should base its decision on the inactive time of each process? As in: Is it orphaned and/or stale?